src/api/app/controllers/webui/workflow_runs_controller.rb

```diff
@@ -1,5 +1,20 @@
 class Webui::WorkflowRunsController < Webui::WebuiController
+  before_action :find_workflow_runs
+
   def index
-    @workflow_runs = WorkflowRunPolicy::Scope.new(User.session, WorkflowRun, { token_id: params[:token_id] }).resolve.page(params[:page])
+    @token = Token::Workflow.find(params[:token_id])
+  end
+
+  def show
+    @workflow_run = @workflow_runs.find(params[:id])
```

Comment:
This is going to run 2 queries:
- it first fetches *all* workflow runs when `resolve` is called inside `find_workflow_runs`
- it then fetches a single workflow run when `find` is called on `@workflow_runs`

We should instead directly fetch the workflow run with its ID instead of using a scope. Scopes are useful, but when fetching multiple records like on index pages. This approach implies changing the `WorkflowRunPolicy`, but it's more much efficient.

Reply:
@dmarcoux added a new pundit policy for it, please have another look